### PR TITLE
2.x Adding ability to configure some transport properties

### PR DESCRIPTION
**Pull request by qiangdavidliu**

### CODE REVIEW

File: eureka2-core/src/main/java/com/netflix/eureka2/config/BasicEurekaRegistryConfig.java [hunk 1]
```diff
@@ -9,32 +11,36 @@
...
+    public static final String EVICTION_STRATEGY_TYPE = StrategyType.PercentageDrop.name();
+    public static final String EVICTION_STRATEGY_VALUE = "20";
 
-    private String evictionTimeoutMs = System.getProperty("eureka.registry.evictionTimeoutMs", EVICTION_TIMEOUT_MS);
-    private String evictionStrategyType = System.getProperty("eureka.registry.evictionStrategy.type", EVICTION_STRATEGY_TYPE);
-    private String evictionStrategyValue = System.getProperty("eureka.registry.evictionStrategy.value", EVICTION_STRATEGY_VALUE);
+    private Long evictionTimeoutMs = Long.parseLong(
+            System.getProperty(evictionTimeoutMsName, EVICTION_TIMEOUT_MS)
+    );
+    private StrategyType evictionStrategyType = StrategyType.valueOf(
```
Comment (tbak) [inline eureka2-core/src/main/java/com/netflix/eureka2/config/BasicEurekaRegistryConfig.java]: We should handle more gracefully errors here, as exceptions are very cryptic when class initialization fails. If property value cannot be parsed, we should log a warning and use a default.

File: eureka2-core/src/main/java/com/netflix/eureka2/config/ConfigNameStrings.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.netflix.eureka2.config;
+
+/**
+ * use camel case for better readability
+ *
+ * @author David Liu
+ */
+public final class ConfigNameStrings {
```
Comment (tbak) [inline eureka2-core/src/main/java/com/netflix/eureka2/config/ConfigNameStrings.java]: Or just ConfigurationNames?

File: eureka2-core/src/main/java/com/netflix/eureka2/config/ConfigNameStrings.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    private static final String EUREKA_PREFIX = "eurekas.";
+
+    public final class Transport {
```
Comment (tbak) [inline eureka2-core/src/main/java/com/netflix/eureka2/config/ConfigNameStrings.java]: TansportNames? So we can do static import, and have a proper context in the class name.

File: eureka2-core/src/main/java/com/netflix/eureka2/config/EurekaTransportConfig.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.netflix.eureka2.config;
+
+import com.netflix.eureka2.transport.EurekaTransports;
+
+/**
+ * @author David Liu
+ */
+public interface EurekaTransportConfig {
```
Comment (tbak) [inline eureka2-core/src/main/java/com/netflix/eureka2/config/EurekaTransportConfig.java]: We need a builder here, that plays well with EurekaBuilder.
For example:
```java
newEurekaBuilder().
    .withTransport(newTransportConfig().withCodec(Codec.Avro))
   .withXXX
```
EurekaBuilder should accept both EurekaTransportConfig and EurekaTransportConfigBuilder for client convenience.

File: eureka2-external-server/src/main/java/com/netflix/eureka2/server/EurekaClientProvider.java [hunk 5]
```diff
@@ -69,7 +69,6 @@ public Integer call(ServerBootstrap server) {
                 }
         );
         return Eureka.newClientBuilder(discoveryResolver, registrationResolver)
-                .withCodec(config.getCodec())
```
Comment (tbak) [inline eureka2-external-server/src/main/java/com/netflix/eureka2/server/EurekaClientProvider.java]: Can we still run EurekaRead server with arbitrary codec? JSON is very useful for debugging purposes.

File: eureka2-core/src/main/java/com/netflix/eureka2/config/BasicEurekaRegistryConfig.java (hunk 1 above)
Comment (qiangdavidliu) [inline eureka2-core/src/main/java/com/netflix/eureka2/config/BasicEurekaRegistryConfig.java]: Will fix.

File: eureka2-core/src/main/java/com/netflix/eureka2/config/ConfigNameStrings.java (hunk 3 above)
Comment (qiangdavidliu) [inline eureka2-core/src/main/java/com/netflix/eureka2/config/ConfigNameStrings.java]: Good suggestion

File: eureka2-core/src/main/java/com/netflix/eureka2/config/EurekaTransportConfig.java (hunk 4 above)
Comment (qiangdavidliu) [inline eureka2-core/src/main/java/com/netflix/eureka2/config/EurekaTransportConfig.java]: There are only two variables right now. We can add a builder if this config grows in the future.

File: eureka2-external-server/src/main/java/com/netflix/eureka2/server/EurekaClientProvider.java (hunk 5 above)
Comment (qiangdavidliu) [inline eureka2-external-server/src/main/java/com/netflix/eureka2/server/EurekaClientProvider.java]: We should still be able to do so, the config just provides a way to specify the codec and defaults to avro, it does not force it to avro.

File: eureka2-core/src/main/java/com/netflix/eureka2/config/EurekaTransportConfig.java (hunk 4 above)
Comment (tbak) [inline eureka2-core/src/main/java/com/netflix/eureka2/config/EurekaTransportConfig.java]: It is more about client API usability. With builder we can have more fluent
object construction.

**Pull request merged**